src/ext/span.h

```diff
@@ -21,19 +20,29 @@ typedef struct ddtrace_span_t {
         uint64_t duration;
     };
     pid_t pid;
-    struct ddtrace_span_t *next;
+};
+typedef struct ddtrace_span_t ddtrace_span_t;
 
-    zend_execute_data *call;
+struct ddtrace_span_fci {
+    zend_execute_data *execute_data;
     struct ddtrace_dispatch_t *dispatch;
+    ddtrace_exception_t *exception;
 #if PHP_VERSION_ID < 70000
+    zval *This;
+    zend_class_entry *called_scope;
+    zend_function *fbc;
+    void **arguments;
     zval *retval;
 #endif
-} ddtrace_span_t;
+    struct ddtrace_span_fci *next;
+    ddtrace_span_t span[1];
```

Comment:
I had some issues with `[1]` even -- since you brought it up I think I'll probably do `ddtrace_span_t span` and stick it directly on the fci, no array-ness, since that didn't work out like I thought it should.